Shaders/FXAA.fx

```diff
@@ -45,18 +61,20 @@ sampler FXAATexture
 {
 	Texture = ReShade::BackBufferTex;
 	MinFilter = Linear; MagFilter = Linear;
-	SRGBTexture = true;
+	#if FXAA_LINEAR_LIGHT
+		SRGBTexture = true;
+	#endif
 };
 
 // Pixel shaders
 
 #if !FXAA_GREEN_AS_LUMA
-float4 FXAALumaPass(float4 vpos : SV_Position, noperspective float2 texcoord : TEXCOORD) : SV_Target
-{
-	float4 color = tex2D(FXAATexture, texcoord.xy);
-	color.a = sqrt(dot(color.rgb, float3(0.299, 0.587, 0.114)));
-	return color;
-}
+	float4 FXAALumaPass(float4 vpos : SV_Position, noperspective float2 texcoord : TEXCOORD) : SV_Target
+	{
+		float4 color = tex2D(ReShade::BackBuffer, texcoord.xy);
+		color.a = sqrt(dot(color.rgb*color.rgb, float3(0.299, 0.587, 0.114)));
```

Comment:
Why use the square of the color?

Reply:
The lightness/luma weights should be multiplied with the color in linear light to be correct. It can be done in gamma light with some error but the speed difference is negligible.

Before, color was decoded with sRGB gamma and then encoded in the alpha channel with 1/2.0 gamma, resulting in slight error in lightness. Now its decoded with 2.0 gamma and encoded with 1/2.0 gamma.

Reply:
Ah. I see. Didn't realize the change in the line right above.